src/cursor/cur_hs.c

```diff
@@ -625,77 +625,27 @@ __curhs_next_visible(WT_SESSION_IMPL *session, WT_CURSOR_HS *hs_cursor)
  */
 int
-__wt_curhs_search_near_before(WT_SESSION_IMPL *session, WT_CURSOR *cursor)
+__wt_curhs_search_near_before(WT_CURSOR *cursor)
 {
-    /*
-     * If the btree id is set alone, it is not possible to position the cursor at a place that is
-     * smaller than set search key, therefore assert that the key must be set to use this function.
-     */
-    WT_ASSERT(session, F_ISSET((WT_CURSOR_HS *)cursor, WT_HS_CUR_KEY_SET));
-    return (__curhs_search_near_helper(session, cursor, true));
+    WT_RET(cursor->bound(cursor, "bound=upper,inclusive=true"));
+    WT_RET(cursor->prev(cursor));
+    WT_RET(cursor->bound(cursor, "action=clear"));
+
+    return (0);
 }
 
 /*
  * __wt_curhs_search_near_after --
  *     Set the cursor position at the requested position or after it.
  */
 int
-__wt_curhs_search_near_after(WT_SESSION_IMPL *session, WT_CURSOR *cursor)
-{
-    return (__curhs_search_near_helper(session, cursor, false));
-}
-
-/*
- * __curhs_search_near_helper --
- *     Helper function to set the cursor position based on search criteria.
- */
-static int
-__curhs_search_near_helper(WT_SESSION_IMPL *session, WT_CURSOR *cursor, bool before)
+__wt_curhs_search_near_after(WT_CURSOR *cursor)
 {
-    WT_DECL_ITEM(srch_key);
-    WT_DECL_RET;
-    int cmp;
-
-    WT_RET(__wt_scr_alloc(session, 0, &srch_key));
-    WT_ERR(__wt_buf_set(session, srch_key, cursor->key.data, cursor->key.size));
-    WT_ERR(cursor->search_near(cursor, &cmp));
-    if (before) {
-        /*
-         * If we want to land on a key that is smaller or equal to the specified key, keep walking
-         * backwards as there may be content inserted concurrently.
-         */
-        if (cmp > 0) {
-            while ((ret = cursor->prev(cursor)) == 0) {
-                WT_STAT_CONN_INCR(session, cursor_skip_hs_cur_position);
-                WT_STAT_DATA_INCR(session, cursor_skip_hs_cur_position);
-                WT_ERR(__wt_compare(session, NULL, &cursor->key, srch_key, &cmp));
-                /*
-                 * Exit if we have found a key that is smaller than or equal to the specified key.
-                 */
-                if (cmp <= 0)
-                    break;
-            }
-        }
-    } else {
-        /*
-         * If we want to land on a key that is larger or equal to the specified key, keep walking
-         * forwards as there may be content inserted concurrently.
-         */
-        if (cmp < 0) {
-            while ((ret = cursor->next(cursor)) == 0) {
-                WT_STAT_CONN_INCR(session, cursor_skip_hs_cur_position);
-                WT_STAT_DATA_INCR(session, cursor_skip_hs_cur_position);
-                WT_ERR(__wt_compare(session, NULL, &cursor->key, srch_key, &cmp));
-                /* Exit if we have found a key that is larger than or equal to the specified key. */
-                if (cmp >= 0)
-                    break;
-            }
-        }
-    }
+    WT_RET(cursor->bound(cursor, "bound=lower,inclusive=true"));
```

Comment:
In order for this to be performant, we need to use precompiled config. See bounded_cursor_perf.cpp for an example.

Reply:
done